back/index.js

```diff
@@ -1,14 +1,36 @@
 const express = require('express');
 const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
 
 const Product = require('./models/product');
 
-const App = express();
-const connect = mongoose.connect('mongodb://localhost:27017/express-react-shop');
+const app = express();
+const port = 5001;
 
-App.get('/products', async (req, res) => {
-  const products = await Product.find();
-  res.json(products);
-});
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.use(bodyParser.json());
 
-App.listen(3000, () => console.log('The server is started'));
+mongoose.connect(
+  'mongodb://localhost:27017/express-react-shop',
```

Comment:
Let's save all important values as environment variables. so Instead of using the database string here, we'll replace it with `process.env.MONGO_URI`. You can use [this library(https://www.npmjs.com/package/dotenv).

This is the best practice for security and for ensuring our application can work in different environments.